modules/text2vec-voyageai/clients/voyageai.go

```diff
@@ -0,0 +1,192 @@
+	}
+
+	body, err := json.Marshal(embeddingsRequest{
+		Input:      input,
+		Model:      model,
+		Truncation: truncation,
+		InputType:  inputType,
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "marshal body")
+	}
+
+	url := v.getVoyageAIUrl(ctx, baseURL)
+	req, err := http.NewRequestWithContext(ctx, "POST", url,
+		bytes.NewReader(body))
+	if err != nil {
+		return nil, errors.Wrap(err, "create POST request")
+	}
+	apiKey, err := v.getApiKey(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "VoyageAI API Key")
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", apiKey))
+	req.Header.Add("Content-Type", "application/json")
+
+	res, err := v.httpClient.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "send POST request")
+	}
+	defer res.Body.Close()
+	bodyBytes, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "read response body")
+	}
+	var resBody embeddingsResponse
+	if err := json.Unmarshal(bodyBytes, &resBody); err != nil {
+		return nil, errors.Wrap(err, "unmarshal response body")
+	}
+
+	if res.StatusCode >= 500 {
+		errorMessage := getErrorMessage(res.StatusCode, resBody.Detail, "connection to VoyageAI failed with status: %d error: %v")
+		return nil, errors.Errorf(errorMessage)
+	} else if res.StatusCode > 200 {
+		errorMessage := getErrorMessage(res.StatusCode, resBody.Detail, "failed with status: %d error: %v")
+		return nil, errors.Errorf(errorMessage)
+	}
+
+	if len(resBody.Data) == 0 || len(resBody.Data[0].Embeddings) == 0 {
+		return nil, errors.Errorf("empty embeddings response")
+	}
+
+	vectors := make([][]float32, len(resBody.Data))
+	for i, data := range resBody.Data {
+		vectors[i] = data.Embeddings
+	}
+
+	return &ent.VectorizationResult{
+		Text:       input,
+		Dimensions: len(resBody.Data[0].Embeddings),
+		Vectors:    vectors,
+	}, nil
+}
+
+func (v *vectorizer) getVoyageAIUrl(ctx context.Context, baseURL string) string {
+	passedBaseURL := baseURL
+	if headerBaseURL := v.getValueFromContext(ctx, "X-VoyageAI-Baseurl"); headerBaseURL != "" {
+		passedBaseURL = headerBaseURL
+	}
+	return v.urlBuilder.url(passedBaseURL)
+}
+
+func getErrorMessage(statusCode int, resBodyError string, errorTemplate string) string {
+	if resBodyError != "" {
+		return fmt.Sprintf(errorTemplate, statusCode, resBodyError)
+	}
+	return fmt.Sprintf(errorTemplate, statusCode)
+}
+
+func (v *vectorizer) getValueFromContext(ctx context.Context, key string) string {
+	if value := ctx.Value(key); value != nil {
+		if keyHeader, ok := value.([]string); ok && len(keyHeader) > 0 && len(keyHeader[0]) > 0 {
+			return keyHeader[0]
+		}
+	}
+	// try getting header from GRPC if not successful
+	if apiKey := modulecomponents.GetValueFromGRPC(ctx, key); len(apiKey) > 0 && len(apiKey[0]) > 0 {
+		return apiKey[0]
+	}
+	return ""
+}
+
+func (v *vectorizer) getApiKey(ctx context.Context) (string, error) {
+	if apiKey := v.getValueFromContext(ctx, "X-VoyageAI-Api-Key"); apiKey != "" {
```

Comment:
you need to change it to `X-Voyageai-Api-Key` - bc that is how golang's http library capitalizes headers